src/huak/project/py_project.rs

```diff
@@ -0,0 +1,201 @@
+        Project {
+            root: path,
+            project_type,
+            config: Config::default(),
+            venv: None,
+        }
+    }
+
+    /// Initialize `Project` from a given path. If a manifest isn't found
+    /// at the path, then we search for a manifest and set the project root
+    /// if it's found.
+    /// ```rust
+    /// use std::env;
+    /// use huak::project::Project;
+    ///
+    /// let cwd = env::current_dir().unwrap();
+    /// let project = Project::from(cwd);
+    /// ```
+    pub fn from(path: PathBuf) -> Result<Project, HuakError> {
+        // TODO: Builder.
+        let config = Config::from(&path)?;
+        let venv = match Venv::from(&path) {
+            Ok(v) => v,
+            Err(HuakError::VenvNotFound) => {
+                Venv::new(path.join(venv::DEFAULT_VENV_NAME))
+            }
+            Err(e) => return Err(e),
+        };
+        let manifest_path = &config.manifest().path;
+
+        // Set the root to the directory the manifest file was found.
+        // TODO: This is probably not the right way to do this.
+        let mut root = path;
+        if let Some(parent) = manifest_path.parent() {
+            root = parent.to_path_buf()
+        }
+
+        // We can't know the project type here, but it probably doesn't matter
+        // much. We'll just use the default.
+        let project_type = ProjectType::default();
+
+        Ok(Project {
+            root,
+            project_type,
+            config,
+            venv: Some(venv),
+        })
+    }
+
+    pub fn create_from_template(&self) -> HuakResult<()> {
+        let name = crate::utils::path::parse_filename(&self.root)?.to_string();
+
+        // Create src subdirectory
+        fs::create_dir_all(self.root.join("src"))?;
+
+        if self.project_type == ProjectType::Library {
+            fs::write(
+                &self.root.join("src").join("__init__.py"),
+                "from .my_math import add_one, times_two\n",
+            )?;
+
+            fs::write(&self.root.join("src").join("my_math.py"),
+                      "# welcome to Huak's sample Python library.\ndef add_one(my_number):\
+                      \n\treturn my_number + 1\n\n\ndef times_two(my_number):\n\treturn my_number * 2\n")?;
+
+            fs::write(&self.root.join("test.py"),
+                      "import unittest\nfrom src import add_one\n\n\n\
+                      class TestSum(unittest.TestCase):\n\tdef test_lib_function(self):\n\t\t\
+                      self.assertEqual(add_one(1), 2)\n\n\nif __name__ == '__main__':\n\tunittest.main()\n")?;
+        } else {
+            fs::create_dir_all(self.root.join("src").join(&name))?;
+            fs::write(&self.root.join("src").join(&name).join("main.py"),
+                      "# This is a sample Python script.\ndef print_hi(name):\
+                      \n\tprint(f'Hello, {name}')\n\n\nif __name__ == '__main__':\n\tprint_hi('World')\n")?;
```

Comment:
hmm I'm not sure I think we should add code that will always be removed. i want the startup step of the workflow to be a seamless get started step. If we do this every time a user bootstraps they'll need to remove this code.

the difference with the app project type is that we're taking the opinion that an "application" project type is typically "executed". so bootstrapping with a blank
```py
def main():
   ...
```
gives the user something to start using right away. it also demos one reason for using the project type to begin with.

Reply:
I think with cargo it just adds a hello world print. I'm not opposed to doing that, though it's one of the very few behaviors I dislike about cargo. I'd personally prefer the equivalent of a `todo!()` in python which would be `...`.

Reply:
hmm not sure I understand the difference. It looks like I'm using Cargo as an example too much, which I'm ok if we differ from cargo, obviously, it's an entire different programming language we're targeting. But I would say, where we differ from a very successful tool we're taking inspiration from, we should probably ask ourselves "why?"

Here's the pretty version of what's in each template.
The idea being, when people first try Huak, they immediately have something that's runnable in both cases (lib and app).
for app they could do
```bash
$ huak new my_project && cd ./my_project
$ huak run
Hello World!
``` 
or if it's a library
```bash
$ huak new my_project --lib && cd ./my_project
$ huak test
----------------------------------------------------------------------
Ran 1 test in 0.001s

OK
``` 

## app template
Again, just mimicking cargo, we give them the classic "hello world." python script.
```python
# This is a sample Python script.
def print_hi(name):
        print(f'Hello, {name}')


if __name__ == '__main__':
        print_hi('World')
```

## library template
Also like cargo, we use a basic library and a basic test suite.
### my_math.py
```python
# welcome to Huak's sample Python library.
def add_one(my_number): 
        return my_number + 1


def times_two(my_number):
        return my_number * 2
```
### test.py
```python
import unittest
from src import add_one


class TestSum(unittest.TestCase):
        def test_lib_function(self):
                self.assertEqual(add_one(1), 2)


if __name__ == '__main__':
        unittest.main()
```

My thinking is, this provides a very welcome experience for people trying Huak for the first time.

Reply:
hmm i dont want to nit this at this stage. what if we start by doing the following and then flesh this out as we get some feedback from users

1. set up app type projects with the main function as an entry point. so something like
```py
def main():
    print("Hello, world!")


if __name__ == "__main__":
    main()
```

2. reduce the lib bootstrap test complexity by either just validating `__version__` exists[1] or leaving test.py with a self-contained test[2]

[1]
*test.py*
```py
from my_package import __version__


def test_version():
    __version__
```

[2]
*test.py*
```py
def test_add():
    res = 2 + 2
    assert res == 4
```

So that ([2]) is equivelent to cargo's self-contained test
```rs
#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        let result = 2 + 2;
        assert_eq!(result, 4);
    }
}
```

The only other comment I'd have is that I'm used to using a tests/ directory for tests. Any thoughts on defaulting to dir vs module?

The fact that you have these opinions is a great thing btw. I hope this doesn't come off as me being combative.

Reply:
> I think with cargo it just adds a hello world print.
oops, left that comment before I saw this

I'm not opposed to cutting down on the amount of python in the starting project, my main sticking point would be something that's immediately runable/testable once bootstrapped. Sounds like we're on the same page in that regard.

Also, sorry I forgot to mention I changed the default huak new to "app" type. Should have mentioned that. I'll take a look at the conversation in that PR.

Reply:
> The fact that you have these opinions is a great thing btw. I hope this doesn't come off as me being combative.

Not at all, I honestly just took most of these implementations/defaults for granted. differing opinions are good stuff, some of the best teams I've been on were people who thought totally different from each other.

Reply:
I'm huuuuuuge on variety in opinions